cmd/thermal-recorder/throttledRecorder.go

```diff
@@ -0,0 +1,134 @@
+package main
+
+import (
+	"log"
+
+	"github.com/TheCacophonyProject/lepton3"
+)
+
+type ThrottledRecorder struct {
+	recorder              Recorder
+	mainBucket            TokenBucket
+	sparseBucket          TokenBucket
+	recording             bool
+	minRecordingLength    uint32
+	sparseRecordingLength uint32
+	askedToWriteFrame     bool
+	throttledFrames       uint32
+	frameCount            uint32
+}
```

Comment:
Can you please add a little commentary somewhere about how the main and sparse buckets work and how they relate to each other?

I like the idea of doing sparse recordings when throttling is happening but the logic is complicated enough to warrant some explanation.